hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/OwnerOfPrecompile.java

```diff
@@ -0,0 +1,67 @@
+package com.hedera.services.store.contracts.precompile.impl;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ *
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.state.submerkle.ExpirableTxnRecord;
+import com.hedera.services.store.contracts.WorldLedgers;
+import com.hedera.services.store.contracts.precompile.SyntheticTxnFactory;
+import com.hedera.services.store.contracts.precompile.codec.DecodingFacade;
+import com.hedera.services.store.contracts.precompile.codec.EncodingFacade;
+import com.hedera.services.store.contracts.precompile.utils.PrecompilePricingUtils;
+import com.hedera.services.store.models.NftId;
+import com.hederahashgraph.api.proto.java.TokenID;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import org.apache.tuweni.bytes.Bytes;
+
+import java.util.function.UnaryOperator;
+
+import static com.hedera.services.exceptions.ValidationUtils.validateTrueOrRevert;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_NFT_SERIAL_NUMBER;
+
+public class OwnerOfPrecompile extends AbstractReadOnlyPrecompile {
```

Comment:
Q: It's unclear from the name that `OwnerOfPrecompile` has anything to do with Nfts.
Based on the naming I wouldn't even assume token logic to be in it as it could be a general contract. 
Is this a result of the only precompile contract currently being the HTS precompile?

Also does this class never apply to a fungible token case?

Reply:
A: ownerOf is a standard function from [ERC721 interface](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/IERC721.sol#L39). And it can`t be applied to fungible token.
Originally this class existed in the HTSPrecompiledContract.java as an inner class.

Reply:
The owner concept can be applied outside of ERC721 is my point but sure if the clarity exists in ownerOf vs owner then that's fine for now.